v2/contribute/decisions/oauth/0018-payload-and-lifetime-after-refresh.mdx

```diff
@@ -0,0 +1,49 @@
+---
+id: "0018"
+title: Token payloads and lifetimes are regenerated on the BE for refresh
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Token payloads and lifetimes are regenerated on the BE for refresh
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+When issuing tokens using a refresh token, we need to decide where we get the payloads and lifetimes from.
+
+## Considered Options
+
+* Token payloads and lifetimes are regenerated on the BE for refresh
+* They are re-used from when the tokens where first issued
+
+## Decision Outcome
+
+Chosen option: Token payloads and lifetimes are regenerated on the BE for refresh
+
```

Comment:
related to this, will the id token's lifetime and custom payload also come in a similar way? Cause then, the API spec for the core needs to all allow to pass in idTokenLifetime and customIDTokenPayload as an input to token token POST API (similar to how we have accessTokenLifetime and customAccessTokenPayload)

Reply:
Custom id-token payload is already there, and we can add a separate lifetime for the id token lifetime as well if you prefer. The basic idea was that it'd get the same lifetime as the access token, but I guess the name is not as descriptive as it could be in this case.

Reply:
We've decided that the id-token will be created separately, so it's lifetime will be controlled by the BE exclusively.